### PR TITLE
Implement Gerrit SubmitChange function for Tide

### PR DESCRIPTION
**Pull request by chaodaiG**
The other part of Tide was implemented, also implement this part

/cc @cjwagner @listx

### CODE REVIEW

File: prow/gerrit/client/client.go
Comment (cjwagner) [inline prow/gerrit/client/client.go:408]: Submitting the exact revision seems like a better option to avoid races: https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#submit-revision

File: prow/gerrit/client/client.go
Comment (cjwagner) [inline prow/gerrit/client/client.go:416]: What does `WaitForMerge` do? I don't see this field documented in the API spec or the go-github library: https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#submit-input
Does it cause the call to block until the merge is complete or does it mean something else?

**Review by cjwagner — APPROVED** [review]
Looks good enough to try out to me. I think using the SubmitRevision endpoint rather than SubmitChange is worthwhile, but not urgent.
/hold

File: prow/gerrit/client/client.go
Comment (chaodaiG) [reply prow/gerrit/client/client.go:416]: It's documented here https://pkg.go.dev/github.com/andygrunwald/go-gerrit#SubmitInput

Comment (chaodaiG) [reply prow/gerrit/client/client.go:416]: Seems like from an old api https://gerrit-documentation.storage.googleapis.com/Documentation/2.11/rest-api-changes.html#submit-change, this is probably deprecated according to https://javadoc.io/static/com.google.gerrit/gerrit-extension-api/3.4.0/deprecated-list.html

Comment (chaodaiG) [reply prow/gerrit/client/client.go:416]: go-gerrit still has it at HEAD though https://github.com/andygrunwald/go-gerrit/blob/41749a62809298f29d8bdf8b21d946853972ec44/changes.go#L207

File: prow/gerrit/client/client.go
Comment (chaodaiG) [reply prow/gerrit/client/client.go:408]: this is not supported yet: https://github.com/andygrunwald/go-gerrit/blob/41749a62809298f29d8bdf8b21d946853972ec44/changes_revision.go#L649.

Did a line by line comparison between Submit Revision and Submit Change, they are pretty much identical other than one thing that cause my attention:
```
 "revision 674ac754f91e64a0efb8087e59a176484bd534d1 is not current revision"
```

So "Submit Revision" technically is safer to avoid submitting a PR that just has a new revision pushed. However afaict Gerrit revokes `Verified` label when a new revision is pushed, which will also prevents the submission according to doc:
```
If the revision cannot be submitted, e.g. because the submit rule doesn’t allow submitting the revision or the revision is not the current revision, the response is “409 Conflict” and the error message is contained in the response body.
```

**Review by chaodaiG — COMMENTED** [review]
/unhold

**Pull request merged**
